src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -5154,18 +5154,48 @@ public static class Extensions
 """;
         var comp = CreateCompilation(src);
         comp.VerifyEmitDiagnostics(
-            // (5,31): error CS9293: Cannot use extension parameter 'object o' in this context.
+            // (5,31): error CS9347: Static members cannot refer to extension parameter 'object o'. Use an instance member instead.
             //         static object M1() => o;
-            Diagnostic(ErrorCode.ERR_InvalidExtensionParameterReference, "o").WithArguments("object o").WithLocation(5, 31),
-            // (6,37): error CS9293: Cannot use extension parameter 'object o' in this context.
+            Diagnostic(ErrorCode.ERR_ExtensionParameterInStaticContext, "o").WithArguments("object o").WithLocation(5, 31),
+            // (6,37): error CS9347: Static members cannot refer to extension parameter 'object o'. Use an instance member instead.
             //         static object M2() { return o; }
-            Diagnostic(ErrorCode.ERR_InvalidExtensionParameterReference, "o").WithArguments("object o").WithLocation(6, 37),
-            // (7,29): error CS9293: Cannot use extension parameter 'object o' in this context.
+            Diagnostic(ErrorCode.ERR_ExtensionParameterInStaticContext, "o").WithArguments("object o").WithLocation(6, 37),
+            // (7,29): error CS9347: Static members cannot refer to extension parameter 'object o'. Use an instance member instead.
             //         static object P1 => o;
-            Diagnostic(ErrorCode.ERR_InvalidExtensionParameterReference, "o").WithArguments("object o").WithLocation(7, 29),
-            // (8,41): error CS9293: Cannot use extension parameter 'object o' in this context.
+            Diagnostic(ErrorCode.ERR_ExtensionParameterInStaticContext, "o").WithArguments("object o").WithLocation(7, 29),
+            // (8,41): error CS9347: Static members cannot refer to extension parameter 'object o'. Use an instance member instead.
             //         static object P2 { get { return o; } }
-            Diagnostic(ErrorCode.ERR_InvalidExtensionParameterReference, "o").WithArguments("object o").WithLocation(8, 41)
+            Diagnostic(ErrorCode.ERR_ExtensionParameterInStaticContext, "o").WithArguments("object o").WithLocation(8, 41)
+            );
+    }
+
+    [Fact]
+    public void ExtensionParameterInStaticContext_WithDifferentContexts()
+    {
+        var src = """
+static class Extensions
+{
+    extension(int p)
+    {
+        // CS9347: Static member of same extension
+        static int M1() => p;
+        
+        // CS9293: Default parameter value  
+        void M3(int x = p) { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (6,28): error CS9347: Static members cannot refer to extension parameter 'int p'. Use an instance member instead.
```

Comment:
>int p [](http://example.com/codeflow?start=89&length=5)

nit: not sure we need to include the type here, the parameter name seems sufficient